data/transactions/logic/assembler.go

```diff
@@ -1857,6 +1857,9 @@ func splitTokens(tokens []string) (current, rest []string) {
 
 // assemble reads text from an input and accumulates the program
 func (ops *OpStream) assemble(text string) error {
+	if strings.TrimSpace(text) == "" {
+		ops.errorf("Cannot assemble empty program text")
```

Comment:
return here?

Reply:
I didn't want to return right away because I want to let the assembler do it's thing and report number of errors. I.e., if I return right away instead of getting the following message:
```
clear_state_program.teal: 0: Cannot assemble empty program text
clear_state_program.teal: 1 error
```

I get the following instead:
```
clear_state_program.teal: 0: Cannot assemble empty program text
clear_state_program.teal: 0: Cannot assemble empty program text
```

Reply:
It should be ok to:
`return ops.errorf("Cannot assemble empty program text")`
since that's how the Version check works a couple lines lower.
`return ops.errorf("Can not assemble version %d", ops.Version)`
That must show the same problem in the context you're looking at.

Perhaps we can change `ops.ReportProblems` to only report the underlying errors if there's more than 1? Rename to `ops.MultpleErrors`?

Reply:
This worked with the following error goal messages:
```
clear_state_program.teal: 1 error
clear_state_program.teal: 0: Cannot assemble empty program text
```

Reply:
After modifying as above, `test/scripts/e2e_subs/e2e-teal.sh` started failing as the original error wasn't reported for `goal clerk compile` a logic sig (bad compiler version). So I modified `ops.MultipleErrors()` to also report the original unique error along with the count in the same line. Now the error looks like:

```sh
clear_state_program.teal: 1 error: 0: Cannot assemble empty program text
clear_state_program.teal: 0: Cannot assemble empty program text
```